pandas/core/indexes/datetimelike.py

```diff
@@ -602,6 +604,17 @@ def delete(self, loc):
         arr = type(self._data)._simple_new(new_i8s, dtype=self.dtype, freq=freq)
         return type(self)._simple_new(arr, name=self.name)
 
+    @doc(Index._convert_arr_indexer)
+    def _convert_arr_indexer(self, keyarr):
+        if lib.infer_dtype(keyarr) == "string":
+            # Weak reasoning that indexer is a list of strings
+            # representing datetime or timedelta or period
+            extension_arr = pd_array(keyarr, self.dtype)
+            converted_arr = extract_array(extension_arr, extract_numpy=True)
```

Comment:
im a little confused.  do we want to end up with a StringArray here or DatetimeArray/PeriodArray?

Reply:
oh never mind, didnt notice self.dtype arg the first time around